File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        n = len(height)
        l,r =0,n-1
        maxArea = 0
        while l<r:
            maxArea = max(maxArea, min(height[r],height[l]) * (r-l))
            if height[l] > height[r]:
                r-=1
            else:
                l+=1
        return maxArea

'''
    two pointer
    area = (r-l)*min height (r,l)
'''

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0, r=height.size()-1;
        int res=0;
        while(l<r) {
            int maxArea = (r-l)*min(height[l], height[r]);
            res = max(maxArea, res);
            if(height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return res;
    }
};